src/accumulator/util.rs

```diff
@@ -12,27 +12,62 @@ pub fn is_root_position(position: u64, num_leaves: u64, forest_rows: u8) -> bool
     return root_present && root_pos == position;
 }
 
-// extractTwins is a optimization for batched deletions. It checks if the nodes
-// being deleted also have their sibling being deleted. It returns the parents
-// of the deleted siblings along with nodes that didn't have a sibling
-pub fn extract_twins(nodes: Vec<u64>, forest_rows: u8) -> (Vec<u64>, Vec<u64>) {
-    let mut parents = Vec::new();
-    let mut twined = Vec::new();
+// removeBit removes the nth bit from the val passed in. For example, if the 2nd
+// bit is to be removed from 1011 (11 in dec), the returned value is 111 (7 in dec).
+pub fn remove_bit(val: u64, bit: u64) -> u64 {
+    let mask = ((2 << bit) - 1) as u64;
+    let upper_mask = std::u64::MAX ^ mask;
+    let upper = val & upper_mask;
+
+    let mask = ((1 << bit) - 1) as u64;
+    let lower_mask = !(std::u64::MAX ^ mask);
+    let lower = val & lower_mask;
+
+    ((upper >> 1) | lower) as u64
+}
+pub fn calc_next_pos(position: u64, del_pos: u64, forest_rows: u8) -> Result<u64, String> {
+    let del_row = detect_row(del_pos, forest_rows);
+    let pos_row = detect_row(position, forest_rows);
+
+    if del_row < pos_row {
+        return Err("calcNextPosition fail.".to_string());
```

Comment:
Should mention why we failed in the error message. Should mention that since the passed in `del_pos` is in a lower row than `position`, we've failed.

Reply:
With the change, we just get this as the error message.

```
`Err("calc_next_pos fail: 0 < 1")`'
```

Which is vague since the caller doesn't know what the 0 and 1 mean.

I suggest something like this:

```diff
diff --git a/src/accumulator/util.rs b/src/accumulator/util.rs
index 3504f8c..e43a4df 100644
--- a/src/accumulator/util.rs
+++ b/src/accumulator/util.rs
@@ -30,7 +30,7 @@ pub fn calc_next_pos(position: u64, del_pos: u64, forest_rows: u8) -> Result<u64
     let pos_row = detect_row(position, forest_rows);
 
     if del_row < pos_row {
-        return Err(format!("calc_next_pos fail: {del_row} < {pos_row}"));
+        return Err(format!("calc_next_pos fail: del_pos of {del_pos} is at a lower row than position at {position}"));
     }
 
     // This is the lower bits where we'll remove the nth bit.
```